notebooks/MAST/HSC/CASJOBS/HCV_CASJOBS/HCV_casjobs_demo.ipynb

```diff
@@ -0,0 +1,995 @@
+    "\n",
+    "```\n",
+    "pip install --upgrade git+git://github.com/rlwastro/mastcasjobs@master\n",
+    "```\n",
+    "\n",
+    "### Set up your CasJobs account information\n",
+    "\n",
+    "You must have a [MAST Casjobs][3] account. Note that MAST Casjobs accounts are independent of SDSS Casjobs accounts.\n",
+    "\n",
+    "For easy startup, you can optionally set the environment variables `CASJOBS_USERID` and/or `CASJOBS_PW` with your Casjobs account information.  The Casjobs user ID and password are what you enter when logging into Casjobs.\n",
+    "\n",
+    "This script prompts for your Casjobs user ID and password during initialization if the environment variables are not defined.\n",
+    "\n",
+    "\n",
+    "  [1]: https://www.anaconda.com/products/distribution\n",
+    "  [2]: https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/channels.html\n",
+    "  [3]: https://mastweb.stsci.edu/hcasjobs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import astropy\n",
+    "import time\n",
+    "import sys\n",
+    "import os\n",
+    "import requests\n",
+    "import json\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "from PIL import Image\n",
+    "from io import BytesIO\n",
+    "\n",
+    "from astropy.table import Table, join\n",
+    "\n",
+    "# check that version of mastcasjobs is new enough\n",
+    "# we are using some features not in version 0.0.1\n",
+    "from pkg_resources import get_distribution\n",
+    "from packaging.version import Version as V\n",
+    "\n",
+    "assert V(get_distribution(\"mastcasjobs\").version) >= V('0.0.2'), \"\"\"\n",
+    "A newer version of mastcasjobs is required.\n",
+    "Update mastcasjobs to current version using this command:\n",
+    "pip install --upgrade git+git://github.com/rlwastro/mastcasjobs@master\n",
+    "\"\"\"\n",
+    "\n",
+    "import mastcasjobs\n",
+    "\n",
+    "# set width for pprint\n",
+    "astropy.conf.max_width = 150"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# set universal matplotlib parameters\n",
+    "plt.rcParams.update({'font.size': 16})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "HSCContext = \"HSCv3\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set up Casjobs environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "if not os.environ.get('CASJOBS_USERID'):\n",
+    "    os.environ['CASJOBS_USERID'] = input('Enter Casjobs UserID:')\n",
+    "if not os.environ.get('CASJOBS_PW'):\n",
+    "    os.environ['CASJOBS_PW'] = getpass.getpass('Enter Casjobs password:')"
```

Comment:
This will not work with the CI as-is, *but* I think the solution is to: 1) set up a testing casjobs account with a known userid/pw, 2) install some secrets into the CI environment that populates these two environment variables.

That's easy to do with either circle-ci or github actions, but I think we can delay it until we get the CI itself working.  But I'm leaving the comment here so that we know to do it whenever the time comes!  (should also make an issue to fix it if we merge this before the CI is working again)

Reply:
flagging @rlwastro

Reply:
I do think the approach suggested by @eteq will work on the Casjobs side.  If the CI-side setup is possible, we can create a MAST casjobs test account that is usable here.